academic-services/knowledge-exploration-service/tutorial-grammar-design.md

```diff
@@ -0,0 +1,377 @@
+
+## Prerequisites
+
+- [Microsoft Academic Knowledge Exploration Service (MAKES) subscription](get-started-setup-provisioning.md) (release version after 2020-11-30)
+- Completion of ["Link private publication records with MAKES entities"](tutorial-entity-linking.md) tutorial
+- Completion of ["Build a library browser with contextual filters"](tutorial-schema-design.md) tutorial
+- Read through ["How to define index schema"](how-to-index-schema.md) how-to guide
+- Read through ["How to define grammar"](how-to-grammar.md) how-to guide
+- Download the [sample search schema for linked private library publications](samplePrivateLibraryData.linked.search.schema.json)
+- Download the [sample search grammar for linked private library publications](samplePrivateLibraryData.linked.search.grammar.xml)
+
+## Create a searchable index
+
+In order to search the index, we will have modify the schema from ["Build a library browser with contextual filters"](tutorial-entity-linking.md) tutorial and add index operations on more attributes.
+
+### Modify schema to support search
+
+We modified/added the following attributes from the ["Build a library browser with contextual filters"](tutorial-schema-design.md) tutorial to make the index searchable.
+
+| Attribute Name | Description| Index Data Type | Index Operations |
+| ---- | ---- | ---- | ---- |
+| `Abstract.Words` | A words array containing all distinct words from the publication's abstract. Used for supporting partial matches in abstract search (partial attribute search). | `string*` | `["equals"]` |
+| `DOI` | Indicates that "DOI" attribute is a object composed of multiple attributes | `Composite?` | - |
+| `DOI.Name` | Used for providing direct match in DOI search (attribute search). | `string?` | `["equals"]` |
+| `Title` | Indicates that "Title" attribute is an object composed of multiple attributes | `Composite?` | - |
+| `Title.Name` | Used for providing direct match in title search (attribute search). | `blob?` | - |
+| `Title.Words` | A words array containing all distinct words from the publication's title. Used for supporting partial matches in title search (partial attribute search). | `string*` | `["equals"]` |
+| `FullTextUrl` | Indicates that "FullTextUrl" attribute is an object composed of multiple attributes | `Composite?` | - |
+| `FullTextUrl.Name` | Used for providing direct match in FullTextUrl lookup (attribute search) | `string?` | `["equals"]` |
+
+#### Search attributes
+
+Attributes that we plan to use for processing search queries needs to be indexed, similar to filter attributes in the ["Build a library browser with contextual filters"](tutorial-schema-design.md) tutorial. For example, to enable our application to process search queries like "papers from microsoft about machine learning" by enabling the `equals` index operation on the `FieldsOfStudy.Name` and `AuthorAffiliations.AuthorName` attributes.
```

Comment:
>For example, to enable our application to process search queries like "papers from microsoft about machine learning" by enabling the `equals` index operation on the `FieldsOfStudy.Name` and `AuthorAffiliations.AuthorName` attributes. [](start = 207, length = 233)

For example, to enable our application to process search queries like "papers from microsoft about machine learning" we need to enable the `equals` index operation on the `FieldsOfStudy.Name` and `AuthorAffiliations.AuthorName` attributes.